homeassistant/components/aladdin_connect/cover.py

```diff
@@ -28,25 +26,33 @@ async def async_setup_entry(
     async_add_entities: AddEntitiesCallback,
 ) -> None:
     """Set up the Aladdin Connect platform."""
-    acc: AladdinConnectClient = hass.data[DOMAIN][config_entry.entry_id]
+    session: api.AsyncConfigEntryAuth = config_entry.runtime_data
+    acc = AladdinConnectClient(session)
     doors = await acc.get_doors()
     if doors is None:
         raise PlatformNotReady("Error from Aladdin Connect getting doors")
+    device_registry = dr.async_get(hass)
+    doors_to_add = []
+    for door in doors:
+        existing = device_registry.async_get(door.unique_id)
+        if existing is None:
+            doors_to_add.append(door)
```

Comment:
I am not sure what the idea is here, but currently you only add garage doors that have been added previously. This should be revistied and fixed